persistence/genesis.go

```diff
@@ -0,0 +1,371 @@
+package persistence
+
+import (
+	"encoding/hex"
+	"fmt"
+	"github.com/pokt-network/pocket/persistence/schema"
+	"github.com/pokt-network/pocket/shared/types/genesis"
+	"github.com/pokt-network/pocket/utility/types"
+	"log"
+)
+
+func (pm *persistenceModule) PopulateGenesisState(state *genesis.GenesisState) { // TODO (Andrew) genericize with actors interface once merged with #111
+	log.Println("Populating genesis state...")
+	rwContext, err := pm.NewRWContext(0)
+	if err != nil {
+		log.Fatal(fmt.Sprintf("an error occurred creating the rwContext for the genesis state: %s", err.Error()))
+	}
+	for _, acc := range state.Accounts {
+		err = rwContext.SetAccountAmount(acc.Address, acc.Amount)
+		if err != nil {
+			log.Fatal(fmt.Sprintf("an error occurred inserting an acc in the genesis state: %s", err.Error()))
+		}
+	}
+	for _, pool := range state.Pools {
+		err = rwContext.InsertPool(pool.Name, pool.Account.Address, pool.Account.Amount)
+		if err != nil {
+			log.Fatal(fmt.Sprintf("an error occurred inserting an pool in the genesis state: %s", err.Error()))
+		}
+	}
+	for _, act := range state.Apps {
+		err = rwContext.InsertApp(act.Address, act.PublicKey, act.Output, act.Paused, int(act.Status), act.MaxRelays, act.StakedTokens, act.Chains, act.PausedHeight, act.UnstakingHeight)
+		if err != nil {
+			log.Fatal(fmt.Sprintf("an error occurred inserting an app in the genesis state: %s", err.Error()))
+		}
+	}
+	for _, act := range state.ServiceNodes {
+		err = rwContext.InsertServiceNode(act.Address, act.PublicKey, act.Output, act.Paused, int(act.Status), act.ServiceUrl, act.StakedTokens, act.Chains, act.PausedHeight, act.UnstakingHeight)
+		if err != nil {
+			log.Fatal(fmt.Sprintf("an error occurred inserting a service node in the genesis state: %s", err.Error()))
+		}
+	}
+	for _, act := range state.Fishermen {
+		err = rwContext.InsertFisherman(act.Address, act.PublicKey, act.Output, act.Paused, int(act.Status), act.ServiceUrl, act.StakedTokens, act.Chains, act.PausedHeight, act.UnstakingHeight)
+		if err != nil {
+			log.Fatal(fmt.Sprintf("an error occurred inserting a fisherman in the genesis state: %s", err.Error()))
+		}
+	}
+	for _, act := range state.Validators {
+		err = rwContext.InsertValidator(act.Address, act.PublicKey, act.Output, act.Paused, int(act.Status), act.ServiceUrl, act.StakedTokens, act.PausedHeight, act.UnstakingHeight)
+		if err != nil {
+			log.Fatal(fmt.Sprintf("an error occurred inserting a validator in the genesis state: %s", err.Error()))
+		}
+	}
+	if err = rwContext.InitParams(); err != nil { // TODO (Team) use params from genesis file not hardcoded
+		log.Fatal(fmt.Sprintf("an error occurred initializing params: %s", err.Error()))
+	}
+	if err = rwContext.Commit(); err != nil {
+		log.Fatal(fmt.Sprintf("an error occurred during commit() on genesis state %s ", err.Error()))
+	}
+}
+
+func (p PostgresContext) GetAllAccounts(height int64) (accs []*genesis.Account, err error) {
```

Comment:
I haven't taken a stab at this yet, but I do believe there's an opportunity to genercize this. Potentially a pattern like this:

```go
func (p PostgresContext)[A any] GetAllActors(height int64) (actors []Actor, err error) { 
...
acc := new(Actor)
```

Can you take a first stab/thought at it and if not, I can take a personal attempt at implementing it with a smaller code footprint?

Reply:
I would highly encourage us to wait until we have this:
`// TODO (Andrew) genericize with actors interface once merged with #111`

This way we don't even need generics :)

Let me know if this is an acceptable placeholder

Reply:
#149 ^^